### PR TITLE
Issue #4963: socnav planner-family characterization baseline (cheap-lane worker)

### LINKED ISSUE #4963: test(4770-prep): characterization baseline for socnav.py planner-family adapters (pre-decomposition lock)
## Goal

Lock the observable behavior of the SocNav planner-family adapters in
`robot_sf/planner/socnav.py` **before** any decomposition slice touches the file, so the
follow-up refactor can prove behavior preservation by diffing against these tests.

`socnav.py` is 4,941 LOC and already carries a "compatibility facade" docstring, but it
still contains 11 classes and the `make_*_policy` factories in one module:

- `SocNavPlannerConfig`, `SamplingPlannerAdapter`, `TrivialReferencePlannerAdapter`
- `SocNavPlannerPolicy` / `SocNavBenchComplexPolicy`
- `SocialForcePlannerAdapter`, `ORCAPlannerAdapter`, `HRVOPlannerAdapter`
- `_SACADRLModel` / `SACADRLPlannerAdapter` (TensorFlow-gated)
- `PredictionPlannerAdapter`, `SocNavBenchSamplingAdapter`
- factories `make_social_force_policy`, `make_orca_policy`, `make_hrvo_policy`,
  `make_sacadrl_policy`, `make_prediction_policy`

Existing tests (`tests/test_socnav_planner_adapter.py`, `tests/test_socnav_sacadrl_adapter.py`)
cover pieces but there is no dedicated characterization baseline that pins the
public factory + `plan()` outputs across the family.

## Bounded first slice (one PR)

Add `tests/planner/test_socnav_characterization.py` that golden-pins, for each available
adapter family:

1. The public factory (`make_*_policy`) construction and the resulting policy/adapter type.
2. `plan()` / policy call output for a small set of fixed, seeded synthetic observations
   (deterministic RNG seed), asserting exact returned `(v, omega)` (or action array) values.
3. The `diagnostics()` / metadata dict keys that the adapter exposes publicly.
4. Optional-dependency gating: adapters whose backend (`torch`, `tensorflow`, `rvo2`) is
   absent must be `skip`-guarded, not fail — mirror the `try/except ImportError` guards at
   the top of `socnav.py`.

Keep it additive and golden-value focused (same style as
`tests/benchmark/test_metrics_characterization.py`). Do **not** change `socnav.py` in this PR.

## Acceptance criteria

- New file `tests/planner/test_socnav_characterization.py` exists and pins factory + `plan()`
  outputs for at least the ORCA, HRVO, and social-force families (backend-available).
- Backend-gated families (SACADRL/TF, prediction/torch) are `pytest.mark.skipif`-guarded.
- Tests are deterministic across two consecutive local runs (seeded).

## Exact test command

```
uv run pytest tests/planner/test_socnav_characterization.py -q
```

## Behavior preservation

This issue *creates* the lock; nothing to preserve yet. The follow-up decomposition issue
must keep this file **and** the existing `tests/test_socnav_planner_adapter.py`,
`tests/test_socnav_sacadrl_adapter.py`, `tests/test_socnav_observation*.py` green.

## Notes

- Blocks the socnav decomposition refactor (separate follow-up issue; number added after creation).
- Related prior (partial) split: #1634 (CLOSED) extracted only `socnav_occupancy`; the module
  is still a 4.9k-LOC facade.
- Tier 35-45, below fix/evidence work. No metric semantics touched.

Part of #4770

### PR DESCRIPTION
**Pull request by ll7**
## What & why

Implements issue #4963: a **pre-decomposition characterization baseline** that golden-pins the observable behavior of the SocNav planner-family adapters in `robot_sf/planner/socnav.py`, so the follow-up god-class decomposition refactor (#4986) can prove behavior preservation by diffing against these tests.

This PR is **additive test-only** — `socnav.py` is not modified. It mirrors the style of the sibling `tests/benchmark/test_metrics_characterization.py` (#4874) and `tests/planner/test_hybrid_rule_local_planner_characterization.py` (#4964), all part of #4770.

New file: `tests/planner/test_socnav_characterization.py`

## What is pinned

1. **Factory construction + adapter type** — all five families (`make_social_force_policy`, `make_orca_policy`, `make_hrvo_policy`, `make_sacadrl_policy`, `make_prediction_policy`): each returns a `SocNavPlannerPolicy` wrapping the correct adapter type; the policy exposes exactly one `adapter` attribute; default config is a `SocNavPlannerConfig`.
2. **`plan()` exact `(v, omega)` golden values** across four fixed, seeded synthetic scenarios (open corridor, near-goal, head-on pedestrian, goal-to-the-left) for:
   - ORCA heuristic fallback (pure-Python, always available, primary portable lock),
   - HRVO local solver (pure-Python, fully portable),
   - social-force adapter (pysocialforce backend, `skipif`-guarded on `sf_forces`).
3. **Backend-available ORCA rvo2 path** — `skipif`-guarded on `rvo2` (C++ values pinned with `pytest.approx` to tolerate platform float-ordering differences).
4. **Backend-gated families** — SACADRL/TF and prediction/torch are `pytest.mark.skipif`-guarded on their respective backends, mirroring the `try/except ImportError` guards at the top of `socnav.py`; their deterministic fallback `plan()` contracts are pinned when the backend is present.
5. **`diagnostics()` / metadata** — `TrivialReferencePlannerAdapter.diagnostics()` key set and values (`{"adapter","steps","contract"}`), step counting, and the shared `config` metadata surface; plus golden pins for the reference and base `SamplingPlannerAdapter` commands.
6. **Determinism lock** — `test_characterization_is_deterministic_across_instances` asserts two consecutive fresh adapter instances reproduce every golden command.

Design note on portability: HRVO uses its own pure-Python solver (no rvo2), so its values are pinned exactly and are the most portable lock. The ORCA heuristic fallback is also pure-Python and pinned exactly. The rvo2/ORCA and pysocialforce paths are pinned with `pytest.approx` + `skipif` guards so the baseline is robust across environments while still pinning native backend behavior when present.

## Acceptance criteria (from issue)

- [x] New file `tests/planner/test_socnav_characterization.py` exists and pins factory + `plan()` outputs for the ORCA, HRVO, and social-force families.
- [x] Backend-gated families (SACADRL/TF, prediction/torch) are `pytest.mark.skipif`-guarded.
- [x] Tests are deterministic across two consecutive local runs (seeded).
- [x] Additive golden-value focused; no changes to `socnav.py`.

## Validation

Exact test command from the issue, run in a fresh worktree-local `.venv` (`uv sync --all-extras`):

```
uv run pytest tests/planner/test_socnav_characterization.py -q
```

Result — two consecutive runs (determinism acceptance criterion):
```
RUN 1: 34 passed in 3.38s
RUN 2: 34 passed in 1.22s
```

Lint/format on the changed file:
```
uv run ruff check tests/planner/test_socnav_characterization.py   -> All checks passed!
uv run ruff format --check tests/planner/test_socnav_characterization.py  -> 1 file already formatted
```

No regression: the existing suite `tests/test_socnav_planner_adapter.py` → `52 passed` in the same fresh venv.

## Successor / scope

Closes #4963. This is the complete issue (test-only, additive); no further slices remain for this issue. This PR does not duplicate any prior PR — the issue thread confirms "This issue *creates* the lock; nothing to preserve yet," and no prior PR merged a slice of #4963. It unblocks the socnav facade decomposition (#4986), which must keep this file green.

## Evidence classification

- Result classification: characterization/lock (test-only), no research/benchmark/metric claim.
- Evidence tier: focused CPU test execution in a worktree-local `uv sync --all-extras` venv; no training, Slurm, campaigns, or benchmark/paper claims.
- Mode: native adapter + heuristic/`fallback` fallback paths pinned; `skipif` guards document optional-backend availability. No degraded execution presented as success.

This PR was produced by the Pi-harness/GLM-5.2 cheap implementation lane; the review gate hardens and merges.

### CODE REVIEW

**Comment by ll7** [comment]
## Gate findings — head `e9c41d7621a8da1fe2130e2fbc8ab472b9d239d0`

0 open review threads (including CodeRabbit).

No findings. The additive characterization suite uses in-memory fixtures (no cwd-relative paths), explicitly gates optional backends, and covers the required factory, golden-command, metadata, and repeatability contracts. Focused validation follows before merge.
